plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/SourceSetsModelBuilder.java

```diff
@@ -207,6 +266,36 @@ private String createUniqueDisplayName(DefaultGradleSourceSet sourceSet,
     return usedName;
   }
 
+  private Set<Object> getArchiveSourcePaths(CopySpec copySpec) {
+    Set<Object> sourcePaths = new HashSet<>();
+    if (copySpec instanceof DefaultCopySpec) {
+      DefaultCopySpec defaultCopySpec = (DefaultCopySpec) copySpec;
+      sourcePaths.addAll(defaultCopySpec.getSourcePaths());
+      // DefaultCopySpec#getChildren changed from Iterable to Collection
+      if (GradleVersion.current().compareTo(GradleVersion.version("6.2")) >= 0) {
```

Comment:
What about

```java

Object children = defaultCopySpec.getChildren();
if (children instanceof Collection) {
    ...
} else if (child instanceof Iterable) {
    ...
}
```

Reply:
That would only work if we compiled the project under an earlier Gradle API version which used `Iterable`.  Which I don't think is a good idea as we'd be stuck on that version and we'd have to use reflection for other newer methods.
With the current version, the `getChildren` method returns a `Collection` and this is set in the `SourceSetsModelBuilder` class file.  That means that when it's run under an earlier version it will fail to find this method as the method in the earlier version returns `Iterable`.  Hence the use of reflection.
If you `javap` the `SourceSetsModelBuilder` class you will see what I mean.  Note the expectation of `Collection`...
```
   #262 = Methodref          #238.#263    // org/gradle/api/internal/file/copy/DefaultCopySpec.getChildren:()Ljava/util/Collection;
   #263 = NameAndType        #264:#104    // getChildren:()Ljava/util/Collection;
```

Reply:
Do you have an idea of the oldest version of Gradle you want to support?  The other option is making 6.2 the minimum version but I don't think that's a good idea.  From my experience, a lot of people use older versions of Gradle, partly because they are using plugins that are discontinued and/or don't support later versions of Gradle.

Reply:
> That means that when it's run under an earlier version it will fail to find this method as the method in the earlier version returns Iterable.

You are right. Thank you for the explanation!

> Do you have an idea of the oldest version of Gradle you want to support?

No idea for that. I was just try the best to support a much as I can.

> From my experience, a lot of people use older versions of Gradle

True, data shows that lots of people are using Gradle < 4.0, which is a quite old version.